psychopy/experiment/components/grating/__init__.py

```diff
@@ -160,3 +160,48 @@ def writeInitCode(self, buff):
         depth = -self.getPosInRoutine()
         code += ", depth=%.1f)\n" % depth
         buff.writeIndentedLines(code)
+
+    def writeInitCodeJS(self, buff):
+        # do we need units code?
+        if self.params['units'].val == 'from exp settings':
+            unitsStr = "units : undefined, "
+        else:
+            unitsStr = "units : %(units)s, " % self.params
+
+        # replace variable params with defaults
+        inits = getInitVals(self.params, 'PsychoJS')
+
+        for paramName in inits:
+            val = inits[paramName].val
+            if val is True:
+                inits[paramName] = 'true'
+            elif val is False:
+                inits[paramName] = 'false'
+            elif val in [None, 'None', 'none', '', 'sin']:
+                inits[paramName].valType = 'code'
+                inits[paramName].val = 'undefined'
+
+        code = ("{inits[name]} = new visual.GratingStim({{\n"
+                "  win : psychoJS.window,\n"
```

Comment:
As we write new boilerplate, I think we should try to be consistent in using `buff.setIndentLevel` rather than manually indenting, and using `%(key)s` with `+` signs to add in the `unitsStr` rather than `.format` with multiple dicts

Reply:
I think that requires a discussion where we look at the visuals of the different options and agree together. For now, I think this is easier to read